---
title: "Reproducing the MAFtools MAF summary"
author: "Ryan Morin"
format: html
editor: visual
output: 
  html_document:
    df_print: paged
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  cache = TRUE,
  comment = "#>",
  warning = FALSE,
  message = FALSE
)
```

Prerequisites

```{r load}
#| warning: false
library(GAMBLR.open)
suppressMessages(library(tidyverse))
```

Get some metadata

```{r mymeta}

my_meta <- get_gambl_metadata(seq_type_filter = c("genome","capture")) %>%
  dplyr::filter(
    pathology %in% c("FL", "DLBCL")
  ) %>%
  dplyr::select(-study,-reference_PMID)

group_by(my_meta, seq_type, pathology) %>% count()

my_meta = check_and_clean_metadata(my_meta,duplicate_action = "keep_first")

group_by(my_meta, seq_type, pathology) %>% count()


```

```{r coding}


capture_coding <- get_coding_ssm(
  these_samples_metadata = my_meta,
  projection = "grch37",
  include_silent = FALSE,
  this_seq_type = "capture"
)

nrow(capture_coding)

genome_coding <- get_coding_ssm(
  these_samples_metadata = my_meta,
  projection = "grch37",
  include_silent = FALSE,
  this_seq_type = "genome"
)

nrow(genome_coding)


```

```{r panel1}
make_panel1 = function(maf_data,base_size=5,title=""){
vc_counted  = maf_data %>% 
  group_by(Variant_Classification) %>% 
  count() %>% 
  arrange(n)
vc_counted$Variant_Classification = factor(vc_counted$Variant_Classification,
                                             levels=unique(vc_counted$Variant_Classification))
mut_cols = get_gambl_colours("mutation")
p1 = ggplot(vc_counted,aes(x=n,y=Variant_Classification,fill=Variant_Classification)) + 
  geom_col() + scale_fill_manual(values=mut_cols)+
  theme_Morons(base_size = base_size,my_legend_position = "none")  +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank(),
    axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
        ) +
  ggtitle(title)
p1
}
make_panel1(genome_coding)
#make_panel1(capture_coding)
```

```{r panel2}
make_panel2 = function(maf_data,base_size=5,title=""){
type_counted  = maf_data %>% 
  group_by(Variant_Type) %>% 
  count() %>% 
  arrange(n)
type_counted$Variant_Type = factor(type_counted$Variant_Type,
                                             levels=unique(type_counted$Variant_Type))

mut_cols = c(SNP="purple1",INS="yellow3",DEL="lightblue",DNP="orange","TNP"="lightgreen")
p2 =ggplot(type_counted,aes(x=n,y=Variant_Type,fill=Variant_Type)) + 
  geom_col() + scale_fill_manual(values=mut_cols)+
  theme_Morons(base_size = base_size,my_legend_position = "none")  +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank(),
    axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
        ) +
  ggtitle(title)
  p2
}
make_panel2(genome_coding)
#make_panel2(capture_coding)
  
```

```{r panel3}
make_panel3 = function(maf_data,base_size=5,title=""){
  

comp = function(base){
  chartr("ACTG", "TGAC",base)
}
maf_data = mutate(maf_data,
                       class = case_when(
                         Reference_Allele %in% c("T","C") ~ 
                           paste0(Reference_Allele,
                                  ">",
                                  Tumor_Seq_Allele2),
                         TRUE ~ paste0(comp(Reference_Allele),
                                       ">",
                                       comp(Tumor_Seq_Allele2)))
                       )

class_counted = maf_data %>% dplyr::filter(Variant_Type == "SNP") %>%
  group_by(class) %>% count()
class_counted = mutate(class_counted,class = factor(class,levels=c("C>A","C>G","C>T","T>C","T>A","T>G")))
mut_cols = get_gambl_colours("rainfall")
p3 = ggplot(class_counted,aes(x=n,y=class,fill=class)) + 
  geom_col() + scale_fill_manual(values=mut_cols)+
  theme_Morons(base_size = base_size,my_legend_position = "none")  + 
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank(),
    axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
        ) +
  ggtitle(title)
p3
}
make_panel3(genome_coding)
#make_panel3(capture_coding)
  
```

```{r panel_4}
make_panel4 = function(maf_data,base_size=5,title=""){
  

type_counted  = maf_data %>% 
  group_by(Tumor_Sample_Barcode,Variant_Classification) %>% 
  count() %>% 
  arrange(desc(n))
type_counted$Tumor_Sample_Barcode = factor(type_counted$Tumor_Sample_Barcode,
                                            levels=unique(type_counted$Tumor_Sample_Barcode))

mut_cols = get_gambl_colours("mutation")
p4 = ggplot(type_counted,aes(x=Tumor_Sample_Barcode,y=n,fill=Variant_Classification)) + 
  geom_col() +
  scale_fill_manual(values=mut_cols) +
  
  theme_Morons(base_size = base_size,my_legend_position = "none") + 
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank(),
    axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
        ) +
  ggtitle(title)
  

p4
}
make_panel4(genome_coding)
#make_panel4(capture_coding)
  
```

```{r panel_5}
library(ggbeeswarm)
make_panel5 = function(maf_data,base_size=5,point_size=0.5,title=""){
  type_counted  = maf_data %>% 
  group_by(Tumor_Sample_Barcode,Variant_Classification) %>% 
  count() %>% 
  arrange(desc(n))
vc_counted  = maf_data %>% 
  group_by(Variant_Classification) %>% 
  count() %>% 
  arrange(n)
vc_counted$Variant_Classification = factor(vc_counted$Variant_Classification,
                                             levels=unique(vc_counted$Variant_Classification))
type_counted$Variant_Classification = factor(type_counted$Variant_Classification,
                                             levels=rev(unique(vc_counted$Variant_Classification)))
p5 = ggplot(type_counted,aes(x=Variant_Classification,y=n,colour=Variant_Classification)) + 
  geom_quasirandom(size=point_size) +
  scale_colour_manual(values=mut_cols) +
  scale_y_log10() +
  theme_Morons(base_size = base_size,my_legend_position = "none")  +
  theme(axis.title.y =element_blank(),
        axis.text.x =element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x=element_blank()) +
  ggtitle(title)
p5
}
make_panel5(genome_coding)
#make_panel5(capture_coding)
```

```{r}
make_panel6 = function(maf_data,base_size=6,top=10,title=""){
  type_counted  = maf_data %>% 
  group_by(Hugo_Symbol,Variant_Classification) %>% 
  count() %>% 
  arrange(n)
print(head(type_counted))
print(tail(type_counted))

top_n = group_by(type_counted,Hugo_Symbol) %>%
  summarise(total=sum(n)) %>%
  arrange(desc(total)) %>%
  slice_head(n=top) %>%
  pull(Hugo_Symbol)
print(top_n)
some_type_counted = dplyr::filter(type_counted,Hugo_Symbol %in% top_n)
some_type_counted$Hugo_Symbol = factor(some_type_counted$Hugo_Symbol,
                                            levels=rev(top_n))
p6 = 
  ggplot(some_type_counted,aes(y=Hugo_Symbol,x=n,fill=Variant_Classification)) + 
  geom_col() +
  scale_fill_manual(values=mut_cols) +
  theme_Morons(base_size = base_size,my_legend_position = "none")  +
  theme(axis.text.x=element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x=element_blank()) +
  ggtitle(title)

p6
}
make_panel6(genome_coding,base_size=6)
#make_panel6(capture_coding)
```

```{r}
library(cowplot)
bs = 4
ps =0.1
p1 = make_panel1(genome_coding,base_size = bs,title="Variant Classification")
p2 = make_panel2(genome_coding,base_size = bs,title="Variant Type")
p3 = make_panel3(genome_coding,base_size = bs,title="SNV Class")
p4 = make_panel4(genome_coding,base_size = bs,title="Variants per sample")
p5 = make_panel5(genome_coding,base_size = bs,point_size=ps,title="Variant Classification Summary")
p6 = make_panel6(genome_coding,base_size = bs, title="Top 10 genes")
all_p = cowplot::plot_grid(p1,p2,p3,p4,p5,p6,nrow = 2,ncol=3)
all_p

```

Testing some other visualizations
