<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Genetic classification of tumours • GAMBLR.open</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><script src="genetic_classification_files/libs/quarto-html/popper.min.js"></script><script src="genetic_classification_files/libs/quarto-html/tippy.umd.min.js"></script><link href="genetic_classification_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="genetic_classification_files/libs/quarto-html/light-border.css" rel="stylesheet">
<!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetic classification of tumours">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GAMBLR.open</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/GAMBLR-open.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Guides</h6></li>
    <li><a class="dropdown-item" href="../articles/mutation_summary.html">Exploring simple somatic mutations</a></li>
    <li><a class="dropdown-item" href="../articles/feature_matrix.html">Summarizing mutation status as a feature matrix</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_classification.html">Genetic classification of tumours</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-quarto">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Genetic classification of tumours</h1>





      <div class="d-none name"><code></code></div>
    </div>






    <section class="section level2"><h2 id="tutorial-classifying-tumours">Tutorial: classifying tumours<a class="anchor" aria-label="anchor" href="#tutorial-classifying-tumours"></a>
</h2>
<p>In the <a href="./tutorials/assemble_feature_matrix.html">previous tutorial</a>, we have looked at how the GAMBLR functionality can be used to generate binary matrices. In this tutorial, we will review how the binary matrix generation can be utilized to classify samples according to different classification systems and tumour labels can be inferred to assign tumours into genetic subgroups.</p>
<div>
<blockquote>
<p><strong>Tip</strong></p>
<p>We would not need to assemble the matrix separately - the <code>classify_</code> family of GAMBLR functions will directly support the maf/seg/bedpe data and handle matrix construction for you.</p>
</blockquote>
</div>
<p>The GAMBLR contains <a href="https://github.com/morinlab/GAMBLR.predict" class="external-link">a collection of machine learning algorithms</a> and functions to pre-format inputs for these models. It contains classifiers of <a href="https://doi.org/10.1182/blood.2022016534" class="external-link">Burkitt</a> and <a href="https://doi.org/10.1182/blood.2022018719" class="external-link">Follicular</a> lymphomas originally published, as well as reproduced of DLBCL classification by the groupings of <a href="https://doi.org/10.1038/s41591-018-0016-8" class="external-link">Chapuy et al</a>, <a href="https://doi.org/10.1182/blood.2019003535" class="external-link">Lacy et al</a>, and <a href="https://doi.org/10.1111/bjh.17132" class="external-link">Runge et al</a>.</p>
<section class="level2"><h3 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h3>
<p>First, we will load required packages. Only <code>GAMBLR.open</code> and <code>tidyverse</code> are needed - they both will load for you all required functionality.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GAMBLR.open</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span></code></pre></div>
</div>
<p>Next, we will obtain the data. We would need the metadata, SSM in maf format, CNV in seg format, and SV in bedped format. For the demonstration purposes, we will use the data bundled with GAMBLR to show the classification functionality. Here, we will subset data to a small number of samples and will illustrate the required formatting and minimal required information.</p>
<section class="level3"><h4 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h4>
<p>The metadata is a data frame which contains the column <code>sample_id</code> listing sample ids for tumours to be classified, and column <code>pathology</code> which will dictate sliding threshold for aSHM site annotation when necessary. The column <code>seq_type</code> is not required for classification purposes, but will be kept in this tutorial for the purpose of subsequent retreival of mutation data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load metadata</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gambl_metadata.html">get_gambl_metadata</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seq_type</span> <span class="op">==</span> <span class="st">"genome"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathology</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FL"</span>, <span class="st">"DLBCL"</span>, <span class="st">"BL"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">sample_id</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="va">ungroup</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">study</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DLBCL_Arthur"</span>, <span class="st">"DLBCL_Thomas"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metadata</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">pathology</span><span class="op">)</span></span></code></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 2
  pathology     n
  &lt;chr&gt;     &lt;int&gt;
1 BL          234
2 DLBCL       334
3 FL          219</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Demonstrate the required columns in metadata</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">metadata</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">pathology</span>, <span class="va">seq_type</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></code></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
  sample_id       pathology seq_type
  &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;
1 00-14595_tumorC DLBCL     genome
2 00-14595_tumorD FL        genome
3 00-15201_tumorA DLBCL     genome
4 00-15201_tumorB DLBCL     genome
5 00-26427_tumorA DLBCL     genome
6 00-26427_tumorC DLBCL     genome  </code></pre>
</div>
</div>
</section><section class="level3"><h4 id="ssm-maf-format">SSM (maf format)<a class="anchor" aria-label="anchor" href="#ssm-maf-format"></a>
</h4>
<p>The key data to be provided for the genetic subgroup classification is SSM in maf format. Only a small subset of columns is required to be used as input information, and here we would demonstrate the expected format and minimal required information for SSM.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ssm_by_samples.html">get_ssm_by_samples</a></span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span></span>
<span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">as.data.frame</span></span>
<span></span>
<span><span class="co"># Only these columns are required</span></span>
<span><span class="va">maf_columns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Hugo_Symbol"</span>, <span class="st">"NCBI_Build"</span>,</span>
<span>    <span class="st">"Chromosome"</span>, <span class="st">"Start_Position"</span>, <span class="st">"End_Position"</span>,</span>
<span>    <span class="st">"Variant_Classification"</span>, <span class="st">"HGVSp_Short"</span>,</span>
<span>    <span class="st">"Tumor_Sample_Barcode"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="va">maf</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">all_of</a></span><span class="op">(</span><span class="va">maf_columns</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">maf</span><span class="op">)</span></span></code></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Hugo_Symbol NCBI_Build Chromosome Start_Position End_Position
1    TNFRSF14     GRCh37          1        2488139      2488139
2        SPEN     GRCh37          1       16245921     16245921
3      BRINP3     GRCh37          1      190187637    190187637
4      BRINP3     GRCh37          1      190189661    190189661
5      BRINP3     GRCh37          1      190264257    190264257
6        BTG2     GRCh37          1      203274977    203274977
  Variant_Classification HGVSp_Short Tumor_Sample_Barcode
1      Nonsense_Mutation      p.W12*            01-20260T
2      Missense_Mutation     p.P515R            01-20260T
3                 Intron                        01-20260T
4                 Intron                        01-20260T
5                 Intron                        01-20260T
6                 Intron                        01-20260T</code></pre>
</div>
</div>
</section><section class="level3"><h4 id="cnv-seg-format">CNV (seg format)<a class="anchor" aria-label="anchor" href="#cnv-seg-format"></a>
</h4>
<p>Several classifiers (<a href="https://doi.org/10.1038/s41591-018-0016-8" class="external-link">Chapuy et al</a>, <a href="https://doi.org/10.1182/blood.2019003535" class="external-link">Lacy et al</a>, and <a href="https://doi.org/10.1111/bjh.17132" class="external-link">Runge et al</a>) require a CNV information to be provided, and here we would demonstrate the expected format and minimal required information for CNV in standard seg format.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_cn_segments.html">get_cn_segments</a></span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span></span>
<span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">as.data.frame</span></span>
<span></span>
<span><span class="co"># Only these columns are required</span></span>
<span><span class="va">seg_columns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"ID"</span>, <span class="st">"chrom"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"LOH_flag"</span>, <span class="st">"log.ratio"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">seg</span> <span class="op">&lt;-</span> <span class="va">seg</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">all_of</a></span><span class="op">(</span><span class="va">seg_columns</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">seg</span><span class="op">)</span></span></code></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         ID chrom     start       end LOH_flag log.ratio
1 02-13135T     1     10001    762600        0 0.0000000
2 02-13135T     1    762601 121500000        0 0.0000000
3 02-13135T     1 142600000 161506889        0 0.0000000
4 02-13135T     1 161506890 161652716        0 0.0000000
5 02-13135T     1 161652717 162110568        0 0.7279926
6 02-13135T     1 162110569 162111399        0 0.0000000</code></pre>
</div>
</div>
</section><section class="level3"><h4 id="sv-bedpe-format">SV (bedpe format)<a class="anchor" aria-label="anchor" href="#sv-bedpe-format"></a>
</h4>
<p>Several classifiers require SV data to be provided, and here we would demonstrate the expected format and minimal required information for SV in standard bed format.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bedpe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_manta_sv.html">get_manta_sv</a></span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span></span>
<span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">as.data.frame</span></span>
<span></span>
<span><span class="co"># Only these columns are required</span></span>
<span><span class="va">bedpe_columns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bedpe</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">11</span><span class="op">]</span></span>
<span></span>
<span><span class="va">bedpe</span> <span class="op">&lt;-</span> <span class="va">bedpe</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">all_of</a></span><span class="op">(</span><span class="va">bedpe_columns</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">bedpe</span><span class="op">)</span></span></code></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  CHROM_A   START_A     END_A CHROM_B   START_B     END_B
1       1 161658631 161658631       3  16509907  16509907
2       1 161663959 161663959       9  37363320  37363320
3       1 161663959 161663959       9  37363320  37363320
4      11  65267283  65267283      14 106110907 106110907
5      11  65267422  65267422      14 106110905 106110905
6      13  91976545  91976545      14 106211857 106211857
                     manta_name SCORE STRAND_A STRAND_B tumour_sample_id
1      MantaBND:21171:0:1:0:0:0   133        +        +         FL2002T1
2     MantaBND:206628:0:1:0:0:0   122        +        +  09-15842_tumorA
3     MantaBND:195941:0:1:0:0:0   151        +        +  09-15842_tumorB
4   MantaBND:152220:0:1:0:0:0:0    88        +        -        15-38154T
5   MantaBND:152220:0:1:0:0:0:0   135        -        +        15-38154T
6 MantaBND:18:59794:59817:0:1:0    90        -        +        15-31924T</code></pre>
</div>
</div>
<div>
<blockquote>
<p><strong>Note</strong></p>
<p>The column <code>manta_name</code> can be any other column from the tool of choice that generated bedpe output and reports the unique id of the SV event, and is expected to be unique for the SV event.</p>
</blockquote>
</div>
</section></section><section class="level2"><h3 id="classify-dlbcl">Classify DLBCL<a class="anchor" aria-label="anchor" href="#classify-dlbcl"></a>
</h3>
<p>Here, we will explore the reproduced DLBCL classification by the groupings of <a href="https://doi.org/10.1038/s41591-018-0016-8" class="external-link">Chapuy et al</a>, <a href="https://doi.org/10.1182/blood.2019003535" class="external-link">Lacy et al</a>, and <a href="https://doi.org/10.1111/bjh.17132" class="external-link">Runge et al</a>. The classification algorithm can be easily controlled by switching the argument <code>method</code> of the <code>classify_dlbcl()</code> function.</p>
<div>
<blockquote>
<p><strong>Important</strong></p>
<p>The DLBCL classifiers were not released with the publications, and the models provided here are the best attempt on reproducing the original models. While the model described by Chapuy et al is reproduced with &gt; 92% accuracy, the Lacy and HMRN classifier results need to be taken with caution as the accuracy of the bundled model is only 80%.</p>
</blockquote>
</div>
<section class="level3"><h4 id="chapuy-et-al--c0-c5-clusters">Chapuy et al. (C0-C5 clusters)<a class="anchor" aria-label="anchor" href="#chapuy-et-al--c0-c5-clusters"></a>
</h4>
<p>The paper published originally in 2018 by <a href="https://doi.org/10.1038/s41591-018-0016-8" class="external-link">Chapuy et al</a> was the first attempt to use systematic approach and classify DLBCL patients into genetic subgroupings using genomic information.</p>
<div>
<blockquote>
<p><strong>Note</strong></p>
<p>This model classifies tumours according to the original 2018 publication, and is not directly related to the 2024 DLBCLass model.</p>
</blockquote>
</div>
<p>Here is how these subgroupings can be recapitulated with GAMBLR functionality:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">predictions</span> <span class="op">&lt;-</span> <span class="fu">classify_dlbcl</span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathology</span> <span class="op">==</span> <span class="st">"DLBCL"</span><span class="op">)</span>,</span>
<span>    maf_data <span class="op">=</span> <span class="va">maf</span>,</span>
<span>    seg_data <span class="op">=</span> <span class="va">seg</span>,</span>
<span>    sv_data <span class="op">=</span> <span class="va">bedpe</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<p>The <code>classify_</code> family of functions by default return both the assembled matrix and predictions. The model reports the confidence of each tumour’s subgoup vote, as well as final label.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># assembled matrix</span></span>
<span><span class="va">predictions</span><span class="op">$</span><span class="va">chapuy_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># subgroup assignment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">chapuy_predictons</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">chapuy_predictons</span>, <span class="va">Chapuy_cluster</span><span class="op">)</span></span></code></pre></div>
</div>
</section><section class="level3"><h4 id="lacy-et-al-">Lacy et al.<a class="anchor" aria-label="anchor" href="#lacy-et-al-"></a>
</h4>
<p>The paper published originally in 2020 by <a href="https://doi.org/10.1182/blood.2019003535" class="external-link">Lacy et al</a> did not release the classification algorithm, but we recapitulated it by training random forest model with 80% accuaracy. Here is how these subgroupings can be recapitulated with GAMBLR functionality:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">predictions</span> <span class="op">&lt;-</span> <span class="fu">classify_dlbcl</span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathology</span> <span class="op">==</span> <span class="st">"DLBCL"</span><span class="op">)</span>,</span>
<span>    maf_data <span class="op">=</span> <span class="va">maf</span>,</span>
<span>    seg_data <span class="op">=</span> <span class="va">seg</span>,</span>
<span>    sv_data <span class="op">=</span> <span class="va">bedpe</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"lacy"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<p>Similar to the Chapuy et al predictions, we can see both the constructed matrix and the confidence of each tumour’s subgoup vote, as well as final label.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># assembled matrix</span></span>
<span><span class="va">predictions</span><span class="op">$</span><span class="va">lacy_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># subgroup assignment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">lacy_predictons</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">lacy_predictons</span>, <span class="va">Lacy_cluster</span><span class="op">)</span></span></code></pre></div>
</div>
</section><section class="level3"><h4 id="hmrn">HMRN<a class="anchor" aria-label="anchor" href="#hmrn"></a>
</h4>
<p>The paper published by <a href="https://doi.org/10.1111/bjh.17132" class="external-link">Runge et al</a> modified the original Lacy et al. subgoupings to more closely adhere the genetic subgroups to the <a href="https://doi.org/10.1016/j.ccell.2020.03.015" class="external-link">LymphGen</a> classification system. The difference between the Runge and Lacy methods is that tumours with truncating <em>NOTCH1</em> mutation will be assigned to a separate subgrouping category regardless of other genetic alterations present in the same tumour sample. Here is how these subgroupings can be recapitulated with GAMBLR functionality:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">predictions</span> <span class="op">&lt;-</span> <span class="fu">classify_dlbcl</span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathology</span> <span class="op">==</span> <span class="st">"DLBCL"</span><span class="op">)</span>,</span>
<span>    maf_data <span class="op">=</span> <span class="va">maf</span>,</span>
<span>    seg_data <span class="op">=</span> <span class="va">seg</span>,</span>
<span>    sv_data <span class="op">=</span> <span class="va">bedpe</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"hmrn"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<p>Take a look at the resulting output</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># assembled matrix</span></span>
<span><span class="va">predictions</span><span class="op">$</span><span class="va">hmrn_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># subgroup assignment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">hmrn_predictons</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">hmrn_predictons</span>, <span class="va">hmrn_cluster</span><span class="op">)</span></span></code></pre></div>
</div>
</section></section><section class="level2"><h3 id="classify-fl">Classify FL<a class="anchor" aria-label="anchor" href="#classify-fl"></a>
</h3>
<p>Here, we will explore the classification of FL tumours into genetic subgroups with differential propensity for transformation as originally described <a href="https://doi.org/10.1182/blood.2022018719" class="external-link">here</a>. The developed model can be easily accessed with <code>classify_fl()</code> function.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">predictions</span> <span class="op">&lt;-</span> <span class="fu">classify_fl</span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathology</span>  <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FL"</span>, <span class="st">"DLBCL"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    maf_data <span class="op">=</span> <span class="va">maf</span>,</span>
<span>    output <span class="op">=</span> <span class="st">"both"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<p>Similar to DLBCL classifier, we can take a look at the assembled matrix, as well as at the predictions and the confidence of each tumour’s vote:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># assembled matrix</span></span>
<span><span class="va">predictions</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># subgroup assignment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">predictions</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">predictions</span>, <span class="va">is_cFL</span><span class="op">)</span></span></code></pre></div>
</div>
</section><section class="level2"><h3 id="classify-bl">Classify BL<a class="anchor" aria-label="anchor" href="#classify-bl"></a>
</h3>
<p>Here, we will explore the classification of BL tumours into genetic subgroups as originally described in<a href="https://doi.org/10.1182/blood.2022016534" class="external-link">Thomas et al</a>. The developed model reported in this study can be easily accessed with <code>classify_bl()</code> function.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">predictions</span> <span class="op">&lt;-</span> <span class="fu">classify_bl</span><span class="op">(</span></span>
<span>    these_samples_metadata <span class="op">=</span> <span class="va">metadata</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathology</span>  <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BL"</span>, <span class="st">"DLBCL"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    maf_data <span class="op">=</span> <span class="va">maf</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<p>Similar to other classifiers, we can take a look at the assembled matrix, as well as at the predictions and the confidence of each tumour’s vote:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># assembled matrix</span></span>
<span><span class="va">predictions</span><span class="op">$</span><span class="va">matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># subgroup assignment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">predictions</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">predictions</span><span class="op">$</span><span class="va">predictions</span>, <span class="va">BL_subgroup</span><span class="op">)</span></span></code></pre></div>
</div>
<p>That’s it!</p>
<p>Happy GAMBLing!</p>
</section></section><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'light-border',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config);
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script></main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ryan Morin, Kostia Dreval.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
