<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Mutation counts across sliding windows for multiple regions. — calc_mutation_frequency_bin_regions • GAMBLR.open</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Mutation counts across sliding windows for multiple regions. — calc_mutation_frequency_bin_regions"><meta name="description" content="Obtain a long tidy or wide matrix of mutation counts across
sliding windows for multiple regions."><meta property="og:description" content="Obtain a long tidy or wide matrix of mutation counts across
sliding windows for multiple regions."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GAMBLR.open</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GAMBLR-open.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Guides</h6></li>
    <li><a class="dropdown-item" href="../articles/mutation_summary.html">Exploring simple somatic mutations</a></li>
    <li><a class="dropdown-item" href="../articles/feature_matrix.html">Summarizing mutation status as a feature matrix</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_classification.html">Genetic classification of tumours</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Mutation counts across sliding windows for multiple regions.</h1>

      <div class="d-none name"><code>calc_mutation_frequency_bin_regions.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Obtain a long tidy or wide matrix of mutation counts across
sliding windows for multiple regions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">calc_mutation_frequency_bin_regions</span><span class="op">(</span></span>
<span>  regions_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  regions_bed <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  these_samples_metadata <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  these_sample_ids <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  this_seq_type <span class="op">=</span> <span class="st">"genome"</span>,</span>
<span>  maf_data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  projection <span class="op">=</span> <span class="st">"grch37"</span>,</span>
<span>  region_padding <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  drop_unmutated <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  skip_regions <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  only_regions <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  slide_by <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  window_size <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  return_format <span class="op">=</span> <span class="st">"wide"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-regions-list">regions_list<a class="anchor" aria-label="anchor" href="#arg-regions-list"></a></dt>
<dd><p>Named vector of regions in the format
c(name1 = "chr:start-end", name2 = "chr:start-end"). If neither <code>regions</code> nor
<code>regions_bed</code> is specified, the function will use GAMBLR aSHM region information.</p></dd>


<dt id="arg-regions-bed">regions_bed<a class="anchor" aria-label="anchor" href="#arg-regions-bed"></a></dt>
<dd><p>Data frame of regions with four columns (chrom, start, end, name).</p></dd>


<dt id="arg-these-samples-metadata">these_samples_metadata<a class="anchor" aria-label="anchor" href="#arg-these-samples-metadata"></a></dt>
<dd><p>Metadata with at least sample_id column.
If not providing a maf data frame, seq_type is also required.</p></dd>


<dt id="arg-these-sample-ids">these_sample_ids<a class="anchor" aria-label="anchor" href="#arg-these-sample-ids"></a></dt>
<dd><p>Vector of sample IDs. Metadata will be subset to
sample IDs present in this vector.</p></dd>


<dt id="arg-this-seq-type">this_seq_type<a class="anchor" aria-label="anchor" href="#arg-this-seq-type"></a></dt>
<dd><p>Optional vector of seq_types to include in heatmap.
Default "genome". Uses default seq_type priority for samples with &gt;1 seq_type.</p></dd>


<dt id="arg-maf-data">maf_data<a class="anchor" aria-label="anchor" href="#arg-maf-data"></a></dt>
<dd><p>Optional maf data frame. Will be subset to rows where
Tumor_Sample_Barcode matches provided sample IDs or metadata table.
If not provided, maf data will be obtained with get_ssm_by_regions().</p></dd>


<dt id="arg-projection">projection<a class="anchor" aria-label="anchor" href="#arg-projection"></a></dt>
<dd><p>Genome build the function will operate in. Ensure this
matches your provided regions and maf data for correct chr prefix handling. Default "grch37".</p></dd>


<dt id="arg-region-padding">region_padding<a class="anchor" aria-label="anchor" href="#arg-region-padding"></a></dt>
<dd><p>Amount to pad the start and end coordinates by. Default 1000.</p></dd>


<dt id="arg-drop-unmutated">drop_unmutated<a class="anchor" aria-label="anchor" href="#arg-drop-unmutated"></a></dt>
<dd><p>Whether to drop bins with 0 mutations. If returning a
matrix format, this will only drop bins with no mutations in any samples.</p></dd>


<dt id="arg-skip-regions">skip_regions<a class="anchor" aria-label="anchor" href="#arg-skip-regions"></a></dt>
<dd><p>Optional character vector of genes to exclude from the default aSHM regions.</p></dd>


<dt id="arg-only-regions">only_regions<a class="anchor" aria-label="anchor" href="#arg-only-regions"></a></dt>
<dd><p>Optional character vector of genes to include from the default aSHM regions.</p></dd>


<dt id="arg-slide-by">slide_by<a class="anchor" aria-label="anchor" href="#arg-slide-by"></a></dt>
<dd><p>Slide size for sliding window. Default 100.</p></dd>


<dt id="arg-window-size">window_size<a class="anchor" aria-label="anchor" href="#arg-window-size"></a></dt>
<dd><p>Size of sliding window. Default 500.</p></dd>


<dt id="arg-return-format">return_format<a class="anchor" aria-label="anchor" href="#arg-return-format"></a></dt>
<dd><p>Return format of mutations. Accepted inputs are "long" and
"wide". Long returns a data frame of one sample ID/window per row. Wide returns
a matrix with one sample ID per row and one window per column. Using the "wide"
format will retain all samples and windows regardless of the drop_unmutated or
min_count_per_bin parameters. Default wide.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Any additional parameters.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A table of mutation counts for sliding windows across one or more regions. May be long or wide.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function takes a metadata table with <code>these_samples_metadata</code>
parameter and internally calls <code>calc_mutation_frequency_bin_region</code>
(that internally calls <code>get_ssm_by_regions</code>).
to retrieve mutation counts for sliding windows across one or more regions.
May optionally provide any combination of a maf data frame, existing metadata,
or a regions data frame or named vector.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span> <span class="co">#load metadata.</span></span></span>
<span class="r-in"><span> <span class="va">my_meta</span> <span class="op">=</span> <span class="fu"><a href="get_gambl_metadata.html">get_gambl_metadata</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled metadata in GAMBLR.data...</span>
<span class="r-in"><span> <span class="va">dlbcl_bl_meta</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">my_meta</span>, <span class="va">pathology</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DLBCL"</span>, <span class="st">"BL"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co">#get ashm regions</span></span></span>
<span class="r-in"><span> <span class="va">some_regions</span> <span class="op">=</span> <span class="fu">create_bed_data</span><span class="op">(</span><span class="va">grch37_ashm_regions</span>,</span></span>
<span class="r-in"><span>                                fix_names <span class="op">=</span> <span class="st">"concat"</span>,</span></span>
<span class="r-in"><span>                                concat_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>,<span class="st">"region"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span></span></span>
<span class="r-in"><span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">some_regions</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> BED Data Object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genome Build: grch37 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Showing first 10 rows:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    chrom     start       end           name region regulatory_comment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      1   6661482   6662702     KLHL21-TSS    TSS               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      1  23885584  23885835        ID3-TSS    TSS               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      1  28832551  28836339       RCC1-TSS    TSS               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      1  31229012  31232011     LAPTM5-TSS    TSS               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      1 150550814 150552135    MCL1-intron intron               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      2  88904839  88909096 EIF2Ak3-intron intron               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7      2  88925456  88927581    EIF2Ak3-TSS    TSS               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8      2 232572640 232574297       PTMA-TSS    TSS               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9      2 157669490 157671299      FCRL3-TSS    TSS               &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10     1 203274698 203275778    BTG2-intron intron    active_promoter</span>
<span class="r-in"><span> <span class="va">mut_count_matrix</span> <span class="op">&lt;-</span> <span class="fu">calc_mutation_frequency_bin_regions</span><span class="op">(</span></span></span>
<span class="r-in"><span>   these_samples_metadata <span class="op">=</span> <span class="va">dlbcl_bl_meta</span>,</span></span>
<span class="r-in"><span>   regions_bed <span class="op">=</span> <span class="va">some_regions</span></span></span>
<span class="r-in"><span> <span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> id_ease: WARNING! 1783 samples in the provided metadata were removed because their seq types are not the same as in the `set_type` argument. Use `verbose = TRUE` to see their IDs.</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">mut_count_matrix</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]   763 14697</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">mut_count_matrix</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 10</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sample_id `1_6661382` `1_6661482` `1_6661582` `1_6661682` `1_6661782`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> SP59456             0           0           0           0           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> SP59460             0           0           0           0           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> SU-DHL-10           0           0           0           0           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> SU-DHL-4            0           0           0           0           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> Seraphina           0           0           0           0           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> Thomas              0           0           0           0           0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 4 more variables: `1_6661882` &lt;int&gt;, `1_6661982` &lt;int&gt;, `1_6662082` &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   `1_6662182` &lt;int&gt;</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ryan Morin, Kostia Dreval.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

