<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get Coding SSM Status. — get_coding_ssm_status • GAMBLR.open</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Coding SSM Status. — get_coding_ssm_status"><meta name="description" content="Tabulate mutation status (SSM) for a set of genes."><meta property="og:description" content="Tabulate mutation status (SSM) for a set of genes."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GAMBLR.open</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GAMBLR-open.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Guides</h6></li>
    <li><a class="dropdown-item" href="../articles/mutation_summary.html">Exploring simple somatic mutations</a></li>
    <li><a class="dropdown-item" href="../articles/feature_matrix.html">Summarizing mutation status as a feature matrix</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_classification.html">Genetic classification of tumours</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get Coding SSM Status.</h1>

      <div class="d-none name"><code>get_coding_ssm_status.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Tabulate mutation status (SSM) for a set of genes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_coding_ssm_status</span><span class="op">(</span></span>
<span>  <span class="va">gene_symbols</span>,</span>
<span>  <span class="va">these_samples_metadata</span>,</span>
<span>  <span class="va">maf_data</span>,</span>
<span>  include_hotspots <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  keep_multihit_hotspot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  review_hotspots <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  genes_of_interest <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FOXO1"</span>, <span class="st">"MYD88"</span>, <span class="st">"CREBBP"</span><span class="op">)</span>,</span>
<span>  <span class="va">genome_build</span>,</span>
<span>  include_silent <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">include_silent_genes</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gene-symbols">gene_symbols<a class="anchor" aria-label="anchor" href="#arg-gene-symbols"></a></dt>
<dd><p>A vector of gene symbols for which the mutation status
will be tabulated. If not provided, lymphoma genes will be returned
by default.</p></dd>


<dt id="arg-these-samples-metadata">these_samples_metadata<a class="anchor" aria-label="anchor" href="#arg-these-samples-metadata"></a></dt>
<dd><p>The metadata for samples of interest to be
included in the returned matrix. Only the column "sample_id" is
required. If not provided, the example metadata is used as default.</p></dd>


<dt id="arg-maf-data">maf_data<a class="anchor" aria-label="anchor" href="#arg-maf-data"></a></dt>
<dd><p>data frame in maf format. Must be in the grch37 projection.</p></dd>


<dt id="arg-include-hotspots">include_hotspots<a class="anchor" aria-label="anchor" href="#arg-include-hotspots"></a></dt>
<dd><p>Logical parameter indicating whether hotspots object
should also be tabulated. Default is TRUE.</p></dd>


<dt id="arg-keep-multihit-hotspot">keep_multihit_hotspot<a class="anchor" aria-label="anchor" href="#arg-keep-multihit-hotspot"></a></dt>
<dd><p>Logical parameter indicating whether to keep the
gene annotation as mutated when the gene has both hot spot and
non-hotspot mutation. Default is FALSE. If set to TRUE, will report the
number of non-hotspot mutations instead of tabulating for just mutation
presence.</p></dd>


<dt id="arg-review-hotspots">review_hotspots<a class="anchor" aria-label="anchor" href="#arg-review-hotspots"></a></dt>
<dd><p>Logical parameter indicating whether hotspots object
should be reviewed to include functionally relevant mutations or rare
lymphoma-related genes. Default is TRUE.</p></dd>


<dt id="arg-genes-of-interest">genes_of_interest<a class="anchor" aria-label="anchor" href="#arg-genes-of-interest"></a></dt>
<dd><p>A vector of genes for hotspot review. Currently only
FOXO1, MYD88, and CREBBP are supported.</p></dd>


<dt id="arg-genome-build">genome_build<a class="anchor" aria-label="anchor" href="#arg-genome-build"></a></dt>
<dd><p>Reference genome build for the coordinates in the MAF
file. The default is inferred from maf_data.</p></dd>


<dt id="arg-include-silent">include_silent<a class="anchor" aria-label="anchor" href="#arg-include-silent"></a></dt>
<dd><p>Logical parameter indicating whether to include silent
mutations into coding mutations. Default is FALSE.</p></dd>


<dt id="arg-include-silent-genes">include_silent_genes<a class="anchor" aria-label="anchor" href="#arg-include-silent-genes"></a></dt>
<dd><p>Optionally, provide a list of genes for which the
Silent variants to be considered. If provided, the Silent variants for
these genes will be included regardless of the include_silent argument.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Any other parameter. These parameters will be ignored.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame with tabulated mutation status.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function takes a data frame (in MAF-like format) and converts
it to a binary one-hot encoded matrix of mutation status for either a set of
user-specified genes (via gene_symbols) or, if no genes are provided, default
to all lymphoma genes. The default behaviour is to assign each gene/sample_id
combination as mutated only if there is a protein coding mutation for that
sample in the MAF but this can be configured to use synonymous variants in
some (via include_silent_genes) or all (via include_silent) genes.
This function also has other filtering and convenience parameters giving
the user full control of the return. For more information, refer to the
parameter descriptions and examples.
Currently only the grch37 genome build is supported for hotspot annotation
and review for this version of the function.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">coding_tabulated_df</span> <span class="op">=</span> <span class="fu">get_coding_ssm_status</span><span class="op">(</span></span></span>
<span class="r-in"><span> maf_data <span class="op">=</span> <span class="fu"><a href="get_coding_ssm.html">get_coding_ssm</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span> gene_symbols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"EZH2"</span>,<span class="st">"KMT2D"</span>,<span class="st">"CREBBP"</span>,<span class="st">"MYC"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled metadata in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> after linking with metadata, we have mutations from 817 samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled metadata in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(sample_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> annotating hotspots</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(sample_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CREBBPHOTSPOT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> MYD88HOTSPOT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> FOXO1HOTSPOT</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">coding_tabulated_df</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   sample_id KMT2D CREBBP EZH2 MYC CREBBPHOTSPOT MYD88HOTSPOT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                     Akata     0      0    0   1             0            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                       BL2     0      0    0   1             0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                      BL30     1      0    0   1             0            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                      BL41     0      0    0   1             0            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                      BL70     0      0    0   1             0            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 BLGSP-71-06-00001-01A-11D     0      0    0   0             0            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   FOXO1HOTSPOT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#all lymphoma genes from bundled NHL gene list</span></span></span>
<span class="r-in"><span><span class="va">coding_tabulated_df</span> <span class="op">=</span> <span class="fu">get_coding_ssm_status</span><span class="op">(</span></span></span>
<span class="r-in"><span>                           maf_data <span class="op">=</span> <span class="fu"><a href="get_coding_ssm.html">get_coding_ssm</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span>                      <span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled metadata in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> after linking with metadata, we have mutations from 817 samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No gene_symbols provided, defaulting to all lymphoma genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled metadata in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(sample_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> annotating hotspots</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(sample_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CREBBPHOTSPOT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> MYD88HOTSPOT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> FOXO1HOTSPOT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">coding_tabulated_df</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   sample_id TNFRSF14 SPEN CARD11 KMT2D BCL7A FOXO1 BCL2 BTK</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                     Akata        0    1      0     0     0     0    0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                       BL2        0    0      0     0     0     0    0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                      BL30        0    0      0     1     0     0    0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                      BL41        0    0      0     0     0     0    0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                      BL70        0    0      0     0     0     0    0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 BLGSP-71-06-00001-01A-11D        0    0      0     0     0     0    0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   CREBBP</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      0</span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co">#this example would fail because hg38 is not supported by this function (yet)</span></span></span>
<span class="r-in"><span><span class="va">coding_tabulated_df</span> <span class="op">=</span> <span class="fu">get_coding_ssm_status</span><span class="op">(</span>maf_data<span class="op">=</span></span></span>
<span class="r-in"><span>                        <span class="fu"><a href="get_coding_ssm.html">get_coding_ssm</a></span><span class="op">(</span>projection <span class="op">=</span> <span class="st">"hg38"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Error in get_coding_ssm_status(maf_data = get_coding_ssm(projection = "hg38")) : </span></span></span>
<span class="r-in"><span><span class="co"># Currently only grch37 projection (hg19 genome build) is supported.</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ryan Morin, Kostia Dreval.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

