<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get ASHM Count Matrix. — get_ashm_count_matrix • GAMBLR.open</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get ASHM Count Matrix. — get_ashm_count_matrix"><meta name="description" content="Prepare a matrix with one row per sample and one column per
region using a set of hypermutated regions."><meta property="og:description" content="Prepare a matrix with one row per sample and one column per
region using a set of hypermutated regions."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GAMBLR.open</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/GAMBLR-open.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Guides</h6></li>
    <li><a class="dropdown-item" href="../articles/mutation_summary.html">Exploring simple somatic mutations</a></li>
    <li><a class="dropdown-item" href="../articles/feature_matrix.html">Summarizing mutation status as a feature matrix</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_classification.html">Genetic classification of tumours</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get ASHM Count Matrix.</h1>

      <div class="d-none name"><code>get_ashm_count_matrix.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Prepare a matrix with one row per sample and one column per
region using a set of hypermutated regions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_ashm_count_matrix</span><span class="op">(</span></span>
<span>  <span class="va">regions_bed</span>,</span>
<span>  <span class="va">these_samples_metadata</span>,</span>
<span>  <span class="va">this_seq_type</span>,</span>
<span>  projection <span class="op">=</span> <span class="st">"grch37"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-regions-bed">regions_bed<a class="anchor" aria-label="anchor" href="#arg-regions-bed"></a></dt>
<dd><p>A bed file with one row for each region.</p></dd>


<dt id="arg-these-samples-metadata">these_samples_metadata<a class="anchor" aria-label="anchor" href="#arg-these-samples-metadata"></a></dt>
<dd><p>This is used to complete your matrix. All GAMBL
samples will be used by default. Provide a data frame with at least
sample_id for all samples if you are using non-GAMBL data.</p></dd>


<dt id="arg-this-seq-type">this_seq_type<a class="anchor" aria-label="anchor" href="#arg-this-seq-type"></a></dt>
<dd><p>The seq type to return results for. Only used if no
metadata is provided with these_samples_metadata.</p></dd>


<dt id="arg-projection">projection<a class="anchor" aria-label="anchor" href="#arg-projection"></a></dt>
<dd><p>Which genome build to use for the mutations
(must match the coordinate system your regions to avoid a nonsense result)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>matrix</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Values are the number of mutations in that patient in the region.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">regions_bed</span> <span class="op">=</span> <span class="fu">create_bed_data</span><span class="op">(</span><span class="fu">GAMBLR.data</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/GAMBLR.data/man/grch37_ashm_regions.html" class="external-link">grch37_ashm_regions</a></span>,</span></span>
<span class="r-in"><span>                              fix_names<span class="op">=</span><span class="st">"concat"</span>,</span></span>
<span class="r-in"><span>                              concat_cols<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>,<span class="st">"region"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                              sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">my_meta</span> <span class="op">=</span> <span class="fu"><a href="get_gambl_metadata.html">get_gambl_metadata</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pathology</span><span class="op">==</span><span class="st">"DLBCL"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled metadata in GAMBLR.data...</span>
<span class="r-in"><span><span class="va">shm_matrix</span> <span class="op">&lt;-</span> <span class="fu">get_ashm_count_matrix</span><span class="op">(</span></span></span>
<span class="r-in"><span>     regions_bed <span class="op">=</span> <span class="va">regions_bed</span>,</span></span>
<span class="r-in"><span>     this_seq_type <span class="op">=</span> <span class="st">"genome"</span>,</span></span>
<span class="r-in"><span>     these_samples_metadata <span class="op">=</span> <span class="va">my_meta</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> id_ease: WARNING! 1783 samples in the provided metadata were removed because their seq types are not the same as in the `set_type` argument. Use `verbose = TRUE` to see their IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running in default mode of any...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(sample_id, region_name)`</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">shm_matrix</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           AICDA-TSS BACH2-TSS BCL11A-TSS BCL2-TSS BCL2-intron BCL6-Intergenic-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-13135T         0         1          0       20           0                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-20170T         1         1          0        0           0                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-22991T         0         0          0        0           0                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-24937T         0         2         15        4           0                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-28140T         0         0          0        0           0                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-29264T         0         0          0        0           0                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           BCL6-Intergenic-2 BCL6-Intergenic-3 BCL6-Intergenic-4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-13135T                 0                 1                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-20170T                 0                 2                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-22991T                 0                 0                 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-24937T                 0                 1                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-28140T                 0                 0                 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-29264T                 0                 2                 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           BCL6-Intergenic-5 BCL6-TSS BCL7A-TSS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-13135T                 0        4         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-20170T                 0        2         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-22991T                 0        1         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-24937T                 0       12         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-28140T                 0        0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-29264T                 1        3         3</span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co">#this example should fail because the regions_bed is not hg38</span></span></span>
<span class="r-in"><span><span class="va">shm_matrix</span> <span class="op">&lt;-</span> <span class="fu">get_ashm_count_matrix</span><span class="op">(</span>regions_bed<span class="op">=</span><span class="va">regions_bed</span>,</span></span>
<span class="r-in"><span>                            this_seq_type <span class="op">=</span> <span class="st">"genome"</span>,</span></span>
<span class="r-in"><span>                            these_samples_metadata <span class="op">=</span> <span class="va">my_meta</span>,</span></span>
<span class="r-in"><span>                            projection <span class="op">=</span> <span class="st">"hg38"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Error in get_ashm_count_matrix(</span></span></span>
<span class="r-in"><span><span class="co"># Your projection argument does not match the genome_build of regions_bed</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span><span class="co"># format the name column to include the coordinates instead of the gene</span></span></span>
<span class="r-in"><span><span class="va">regions_bed</span> <span class="op">=</span> <span class="fu">create_bed_data</span><span class="op">(</span><span class="fu">GAMBLR.data</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/GAMBLR.data/man/hg38_ashm_regions.html" class="external-link">hg38_ashm_regions</a></span>,</span></span>
<span class="r-in"><span>                           fix_names<span class="op">=</span><span class="st">"concat"</span>,</span></span>
<span class="r-in"><span>                           concat_cols<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr_name"</span>,<span class="st">"hg38_start"</span>,<span class="st">"hg38_end"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                           sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="va">matrix_hg38</span> <span class="op">&lt;-</span> <span class="fu">get_ashm_count_matrix</span><span class="op">(</span>regions_bed<span class="op">=</span><span class="va">regions_bed</span>,</span></span>
<span class="r-in"><span>                                      this_seq_type <span class="op">=</span> <span class="st">"genome"</span>,</span></span>
<span class="r-in"><span>                                      these_samples_metadata <span class="op">=</span> <span class="va">my_meta</span>,</span></span>
<span class="r-in"><span>                                      projection <span class="op">=</span> <span class="st">"hg38"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> id_ease: WARNING! 1783 samples in the provided metadata were removed because their seq types are not the same as in the `set_type` argument. Use `verbose = TRUE` to see their IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running in default mode of any...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(sample_id, region_name)`</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">matrix_hg38</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 2312  129</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">matrix_hg38</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           chr1-203305570-203306650 chr1-226732862-226740184</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-13135T                        0                        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-20170T                        0                        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-22991T                        0                        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-24937T                       21                        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-28140T                        0                        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-29264T                        1                        0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           chr1-226733387-226740281 chr1-28506039-28509827</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-13135T                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-20170T                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-22991T                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-24937T                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-28140T                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-29264T                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           chr1-30756165-30759164 chr11-102317439-102319346</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-13135T                      0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-20170T                      0                         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-22991T                      0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-24937T                      0                         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-28140T                      0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-29264T                      0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           chr11-111377353-111379499 chr11-118883749-118885805</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-13135T                         0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-20170T                         0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 02-22991T                         0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-24937T                         1                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-28140T                         0                         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 04-29264T                         1                         0</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ryan Morin, Kostia Dreval.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

