<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get SSM By Regions. — get_ssm_by_regions • GAMBLR.open</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get SSM By Regions. — get_ssm_by_regions"><meta name="description" content="Efficiently retrieve all mutations across a range of genomic regions."><meta property="og:description" content="Efficiently retrieve all mutations across a range of genomic regions."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GAMBLR.open</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Guides</h6></li>
    <li><a class="dropdown-item" href="../articles/mutation_summary.html">Exploring simple somatic mutations</a></li>
    <li><a class="dropdown-item" href="../articles/feature_matrix.html">Summarizing mutation status as a feature matrix</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_classification.html">Genetic classification of tumours</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get SSM By Regions.</h1>

      <div class="d-none name"><code>get_ssm_by_regions.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Efficiently retrieve all mutations across a range of genomic regions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_ssm_by_regions</span><span class="op">(</span></span>
<span>  <span class="va">these_samples_metadata</span>,</span>
<span>  <span class="va">regions_list</span>,</span>
<span>  <span class="va">regions_bed</span>,</span>
<span>  this_seq_type <span class="op">=</span> <span class="st">"genome"</span>,</span>
<span>  streamlined <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  projection <span class="op">=</span> <span class="st">"grch37"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use_name_column <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  tool_name <span class="op">=</span> <span class="st">"slms-3"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-these-samples-metadata">these_samples_metadata<a class="anchor" aria-label="anchor" href="#arg-these-samples-metadata"></a></dt>
<dd><p>Optional, a metadata table (with sample IDs in a column) to subset the return to.</p></dd>


<dt id="arg-regions-list">regions_list<a class="anchor" aria-label="anchor" href="#arg-regions-list"></a></dt>
<dd><p>A vector of regions in the chr:start-end format to restrict the returned SSM calls to.</p></dd>


<dt id="arg-regions-bed">regions_bed<a class="anchor" aria-label="anchor" href="#arg-regions-bed"></a></dt>
<dd><p>A data frame in BED format with the coordinates you want to retrieve (recommended).
This parameter can also accept an additional column with region names that will be added to the return if <code>use_name_column = TRUE</code></p></dd>


<dt id="arg-this-seq-type">this_seq_type<a class="anchor" aria-label="anchor" href="#arg-this-seq-type"></a></dt>
<dd><p>The this_seq_type you want back, default is genome.</p></dd>


<dt id="arg-streamlined">streamlined<a class="anchor" aria-label="anchor" href="#arg-streamlined"></a></dt>
<dd><p>If set to TRUE (default) only 3 columns will be kept in the returned data frame (start, sample_id and region_name).</p></dd>


<dt id="arg-projection">projection<a class="anchor" aria-label="anchor" href="#arg-projection"></a></dt>
<dd><p>Obtain variants projected to this reference (one of grch37 or hg38), default is grch37.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Set to TRUE to maximize the output to console. Default is TRUE.
This parameter also dictates the verbosity of any helper function internally called inside the main function.</p></dd>


<dt id="arg-use-name-column">use_name_column<a class="anchor" aria-label="anchor" href="#arg-use-name-column"></a></dt>
<dd><p>If your bed-format data frame has a name column (must be named "name") these can be used to name your regions.</p></dd>


<dt id="arg-tool-name">tool_name<a class="anchor" aria-label="anchor" href="#arg-tool-name"></a></dt>
<dd><p>Optionally specify which tool to report variant from. The default is slms-3, also supports "publication" to return the exact variants as reported in the original papers.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Any additional parameters.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a data frame of variants in MAF-like format.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function internally calls get_ssm_by_region to retrieve SSM calls for the specified regions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#basic usage, adding custom names from bundled ashm data frame</span></span></span>
<span class="r-in"><span><span class="va">regions_bed</span> <span class="op">=</span> <span class="fu">create_bed_data</span><span class="op">(</span> <span class="fu">GAMBLR.data</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/GAMBLR.data/man/grch37_ashm_regions.html" class="external-link">grch37_ashm_regions</a></span>,</span></span>
<span class="r-in"><span>                          fix_names <span class="op">=</span> <span class="st">"concat"</span>,</span></span>
<span class="r-in"><span>                          concat_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>,<span class="st">"region"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                          sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">my_meta</span> <span class="op">=</span> <span class="fu"><a href="get_gambl_metadata.html">get_gambl_metadata</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled metadata in GAMBLR.data...</span>
<span class="r-in"><span><span class="co"># get a full MAF-format data frame for all aSHM regions on grch37 coordinates</span></span></span>
<span class="r-in"><span><span class="va">ashm_maf</span> <span class="op">=</span> <span class="fu">get_ssm_by_regions</span><span class="op">(</span>regions_bed <span class="op">=</span> <span class="va">regions_bed</span>,</span></span>
<span class="r-in"><span>                              these_samples_metadata <span class="op">=</span> <span class="va">my_meta</span>,</span></span>
<span class="r-in"><span>                              streamlined <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> id_ease: WARNING! 1838 samples in the provided metadata were removed because their seq types are not the same as in the `set_type` argument. Use `verbose = TRUE` to see their IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running in default mode of any...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">one_region_maf</span> <span class="op">=</span> <span class="fu">get_ssm_by_regions</span><span class="op">(</span>regions_list <span class="op">=</span> <span class="st">"2:136875000-136875097"</span>,</span></span>
<span class="r-in"><span>                         streamlined <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                         projection <span class="op">=</span> <span class="st">"grch37"</span>,</span></span>
<span class="r-in"><span>                         these_samples_metadata <span class="op">=</span> <span class="va">my_meta</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the bundled SSM calls (.maf) calls in GAMBLR.data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> id_ease: WARNING! 1838 samples in the provided metadata were removed because their seq types are not the same as in the `set_type` argument. Use `verbose = TRUE` to see their IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running in default mode of any...</span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># This example fails, as it should</span></span></span>
<span class="r-in"><span><span class="co">#ashm_maf = get_ssm_by_regions(regions_bed = regions_bed,</span></span></span>
<span class="r-in"><span><span class="co">#                              these_samples_metadata = my_meta,</span></span></span>
<span class="r-in"><span><span class="co">#                               projection="hg38")</span></span></span>
<span class="r-in"><span><span class="co"># Error in get_ssm_by_regions(regions_bed = regions_bed, these_samples_metadata = my_meta,  : </span></span></span>
<span class="r-in"><span><span class="co"># requested projection: hg38 and genome_build of regions_bed: grch37 don't match</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ryan Morin, Kostia Dreval.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

